pkg/graph/resolvers/key_contact_test.go

```diff
@@ -0,0 +1,285 @@
+)
+
+// TestCreateKeyContactUser validates creating a user contact for key contact.
+func (suite *ResolverSuite) TestCreateKeyContactUser() {
+	categoryName := "Test Category"
+	newCategory, err := CreateKeyContactCategory(
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		categoryName,
+	)
+	suite.NoError(err)
+
+	userName := "John Doe"
+	subjectArea := "Healthcare"
+	subjectCategoryID := newCategory.ID
+
+	contact, err := CreateKeyContactUser(
+		suite.testConfigs.Context,
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		nil,
+		nil,
+		email.AddressBook{},
+		userName,
+		subjectArea,
+		subjectCategoryID,
+		userhelpers.GetUserInfoAccountInfoWrapperFunc(suite.stubFetchUserInfo),
+	)
+	suite.NoError(err)
+	suite.NotNil(contact)
+	suite.Equal(subjectArea, contact.SubjectArea)
+	suite.Equal(subjectCategoryID, contact.SubjectCategoryID)
+}
+
+// TestCreateKeyContactMailbox validates creating a mailbox contact for key contact.
+func (suite *ResolverSuite) TestCreateKeyContactMailbox() {
+	categoryName := "Test Category"
+	newCategory, err := CreateKeyContactCategory(
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		categoryName,
+	)
+	suite.NoError(err)
+
+	mailboxTitle := "Support Team"
+	mailboxAddress := "support@example.com"
+	subjectArea := "Healthcare"
+	subjectCategoryID := newCategory.ID
+
+	contact, err := CreateKeyContactMailbox(
+		suite.testConfigs.Context,
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		nil,
+		nil,
+		email.AddressBook{},
+		mailboxTitle,
+		mailboxAddress,
+		subjectArea,
+		subjectCategoryID,
+	)
+
+	suite.NoError(err)
+	suite.NotNil(contact)
+	suite.Equal(mailboxTitle, *contact.Name)
+	suite.Equal(mailboxAddress, *contact.Email)
+	suite.Equal(subjectArea, contact.SubjectArea)
+	suite.Equal(subjectCategoryID, contact.SubjectCategoryID)
+}
+
+// TestGetKeyContactByIDLOADER validates fetching a key contact by its ID using the loader.
+func (suite *ResolverSuite) TestGetKeyContactByIDLOADER() {
+	categoryName := "Test Category"
+	newCategory, err := CreateKeyContactCategory(
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		categoryName,
+	)
+	suite.NoError(err)
+
+	// Create a mailbox contact using the existing method
+	mailboxTitle := "Support Team"
+	mailboxAddress := "support@example.com"
+	subjectArea := "Healthcare"
+	subjectCategoryID := newCategory.ID
+
+	contact, err := CreateKeyContactMailbox(
+		suite.testConfigs.Context,
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		nil,
+		nil,
+		email.AddressBook{},
+		mailboxTitle,
+		mailboxAddress,
+		subjectArea,
+		subjectCategoryID,
+	)
+
+	suite.NoError(err)
+
+	// Fetch key contact using the loader
+	keyContact, err := GetKeyContact(suite.testConfigs.Context,
```

Comment:
We have a utility to test data loaders, it's meant to make it fetch multiple things at the same time to validate the result. I'd recommend trying this approach https://github.com/CMS-Enterprise/mint-app/blob/eb527c2de3e3229e613070097f2af60dff95acee/pkg/graph/resolvers/mto_common_solution_contact_test.go#L91.